pkg/controller/nexus/resource/deployment/manager.go

```diff
@@ -125,3 +128,12 @@ func equalPullPolicies(depDeployment, reqDeployment *appsv1.Deployment) bool {
 
 	return depDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy == corev1.PullIfNotPresent
 }
+
+// see: https://github.com/m88i/nexus-operator/issues/156
+// On OpenShift 4.5+ `SecurityContext` is not nil, but a "blank" object.
+// Since we are requesting a nil object in this context, we consider the deployed object to be nil as well.
+func equalizeSecurityContext(depDeployment, reqDeployment *appsv1.Deployment) {
+	if reqDeployment.Spec.Template.Spec.SecurityContext == nil {
+		depDeployment.Spec.Template.Spec.SecurityContext = nil
+	}
+}
```

Comment:
I got a couple of observations about this one.

  1. Once I read the function I understood what's going on, but when I saw its usage in line 94 I didn't understand if this was comparing two SCCs (didn't seem to be that since the parameters are deployments) or what. My understanding is that this ["normalizes" or "canonicalizes"](https://en.wikipedia.org/wiki/Canonicalization)  both possible representations of the SCC withing a deployment, how about naming it that way? `normalizeSecurityContext` or equivalent.
  2. Receiving the deployments makes for additional context, but a more verbose function. If we received the `*SecurityContext` directly, it could be a bit cleaner:

```suggestion
func normalizeSecurityContext(depCtx, reqCtx *corev1.SecurityContext) {
	if reqCtx == nil {
		depCtx = nil
	}
}
```
  Wdyt?

Reply:
I won't change the method signature since we have two `SecurityContext` in the deployment spec. Might cause confusion to the caller. Let's the function decide with one (or ones maybe in the future) to normalize. I wrote the function's name with other thing in mind actually, that's why `equalize`.